### ISSUE TITLE
MPProgressOverlayView (void)hide must be called on main thread

### ISSUE THREAD

**Issue by git-master101** [mopub/mopub-ios-sdk#382]
One line summary of the issue here

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version:
  - GoogleMobileAdsMediationMoPub (5.15.0.0):
    - Google-Mobile-Ads-SDK (>= 7.68.0)
    - mopub-ios-sdk (= 5.15.0)

#### Device model and OS Version:
Any

#### Xcode Version:
12.4

#### Ad Unit IDs used in reproducing the issue:
Not relevant

#### Steps to reproduce the behavior:
A UI Alert was already present while mopub was loading.

#### Expected behavior:
Mopub should configure the ads as normal.

#### Observed behavior:
App crashes because of UI work being called in background thread.

#### Evidence:
```
- (void)hide {
    // Cancel any pending enabling of the close button
    [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(enableCloseButton) object:nil];

    // Hide the close button and make it non-user interactable immediately
    self.closeButton.alpha = 0.0f;
    self.closeButton.hidden = YES;

    // Animate removal of self from the key window
    [UIView animateWithDuration:kProgressOverlayAnimationDuration animations:^{
        self.alpha = 0.0;
    } completion:^(BOOL finished) {
        [self removeFromSuperview];
    }];
}
```
<img width="1534" alt="Screenshot 2021-04-15 at 11 38 41 AM" src="https://user-images.githubusercontent.com/70585569/114848620-28a37180-9ddf-11eb-811f-8ecf4496858a.png">

**Comment by chauduyphanvu** [comment]
@git-master101, thanks for the report. I've relayed it to the rest of the team. Stay tuned for an update!

Vu Chau
MoPub Team

**Comment by git-master101** [comment]
This is the stacktrace from crashlytics

`Crashed: com.apple.root.default-qos
0  libobjc.A.dylib                0x1bf52b234 weak_register_no_lock + 68
1  libobjc.A.dylib                0x1bf52d224 objc_storeWeak + 352
2  UIKitCore                      0x1ad3bb7c0 -[UIViewAnimationState setupWithDuration:delay:view:options:factory:parentState:start:completion:] + 496
3  UIKitCore                      0x1ad3ebea0 +[UIView _setupAnimationWithDuration:delay:view:options:factory:animations:start:animationStateGenerator:completion:] + 528
4  UIKitCore                      0x1ad3ec468 +[UIView(UIViewAnimationWithBlocks) animateWithDuration:animations:completion:] + 92
5  MoPub                          0x10272baac -[MPProgressOverlayView hide] + 180 (MPProgressOverlayView.m:180)
6  MoPub                          0x1026e7168 -[MPAdDestinationDisplayAgent dismissAllModalContent] + 71 (MPAdDestinationDisplayAgent.m:71)
7  MoPub                          0x1026e7078 -[MPAdDestinationDisplayAgent dealloc] + 60 (MPAdDestinationDisplayAgent.m:60)
8  MoPub                          0x10271f3a8 -[MPMoPubNativeAdAdapter .cxx_destruct] + 29 (MPMoPubNativeAdAdapter.m:29)
9  libobjc.A.dylib                0x1bf50db10 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
10 libobjc.A.dylib                0x1bf524840 objc_destructInstance + 80
11 libobjc.A.dylib                0x1bf52b80c _objc_rootDealloc + 80
12 MoPub                          0x10271ee80 -[MPMoPubNativeAdAdapter dealloc] + 141 (MPMoPubNativeAdAdapter.m:141)
13 MoPub                          0x1027216e0 -[MPNativeAd .cxx_destruct] + 62 (MPNativeAd.m:62)
14 libobjc.A.dylib                0x1bf50db10 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
15 libobjc.A.dylib                0x1bf524840 objc_destructInstance + 80
16 libobjc.A.dylib                0x1bf52b80c _objc_rootDealloc + 80
17 MoPub                          0x102720148 -[MPNativeAd dealloc] + 101 (MPNativeAd.m:101)
18 dua                            0x100f04818 -[GADMAdapterMoPub .cxx_destruct] + 20 (GADMAdapterMoPub.m:20)
19 libobjc.A.dylib                0x1bf50db10 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
20 libobjc.A.dylib                0x1bf524840 objc_destructInstance + 80
21 libobjc.A.dylib                0x1bf52b80c _objc_rootDealloc + 80
22 dua                            0x100ed9344 -[GADMediatedAdRenderer .cxx_destruct] + 99 (GADMediatedAdRenderer.m:99)
23 libobjc.A.dylib                0x1bf50db10 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
24 libobjc.A.dylib                0x1bf524840 objc_destructInstance + 80
25 libobjc.A.dylib                0x1bf52b80c _objc_rootDealloc + 80
26 dua                            0x100ed84ac -[GADMediatedAdRenderer dealloc] + 103 (GADMediatedAdRenderer.m:103)
27 libobjc.A.dylib                0x1bf50db10 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
28 libobjc.A.dylib                0x1bf524840 objc_destructInstance + 80
29 libobjc.A.dylib                0x1bf52b80c _objc_rootDealloc + 80
30 libsystem_blocks.dylib         0x1f5f34764 _Block_release + 200
31 libsystem_blocks.dylib         0x1f5f34764 _Block_release + 200
32 libdispatch.dylib              0x1aa0e9db0 _dispatch_client_callout + 20
33 libdispatch.dylib              0x1aa0ecddc _dispatch_queue_override_invoke + 692
34 libdispatch.dylib              0x1aa0fa93c _dispatch_root_queue_drain + 356
35 libdispatch.dylib              0x1aa0fb120 _dispatch_worker_thread2 + 116
36 libsystem_pthread.dylib        0x1f5fa17d8 _pthread_wqthread + 216
37 libsystem_pthread.dylib        0x1f5fa876c start_wqthread + 8`

**Comment by chauduyphanvu** [comment]
This issue has been addressed in [v5.18.0](https://github.com/mopub/mopub-ios-sdk/releases/tag/5.18.0) of the MoPub iOS SDK. Please update and let us know if you continue seeing the issue. Thanks!
